explorer/interpreter/type_checker.cpp

```diff
@@ -649,6 +780,22 @@ auto TypeChecker::ArgumentDeduction(
                                           << "actual: " << *arg;
     }
     const Value* subst_param_type = Substitute(deduced, param);
+    if (param->kind() == Value::Kind::VariableType) {
+      const auto& binding = cast<VariableType>(*param).binding();
+      if (binding.has_static_type()) {
+        if (binding.static_type().kind() != Value::Kind::TypeType &&
+            binding.static_type().kind() != Value::Kind::ConstraintType &&
+            binding.static_type().kind() != Value::Kind::InterfaceType) {
+          if (!IsImplicitlyConvertible(subst_param_type, &binding.static_type(),
+                                       impl_scope, true)) {
```

Comment:
If there is a user-defined conversion here, we'll need to invoke it at some point. But... can we ever need a user-defined conversion here? I think the answer might be no: there are no user-defined type-of-types, so there can be no user-defined conversions on types. Passing `false` for `allow_user_defined_conversions` seems like it might be safer for now.